packages/core/src/validate.ts

```diff
@@ -10,9 +10,10 @@ import { UpgradesError, ErrorDescriptions } from './error';
 import { SrcDecoder } from './src-decoder';
 import { isNullish } from './utils/is-nullish';
 
-export type Validation = Record<string, ValidationResult>;
+export type ValidationLog = Array<RunValidation>;
```

Comment:
These are equivalent but we use this syntax.

```suggestion
export type ValidationLog = RunValidation[];
```